types/sinon/index.d.ts

```diff
@@ -110,6 +114,9 @@ declare namespace Sinon {
         resolves(value?: any): SinonStub;
         throws(type?: string): SinonStub;
         throws(obj: any): SinonStub;
+        throwsArg(index: number): SinonStub;
+        throwsException(type?: string): SinonStub;
```

Comment:
I guess you're referring to both the previously existing `throws(type?: string)` and the newly added `throwsException(type?: string)`.

In which case, the referenced issue hasn't resulted in any changes yet. So whether or not it gets changed in future shouldn't matter as right now it is part of the API. We shouldn't be ignoring parts of the library because we expect it to be removed in future, as people may well be using it still.